tests/ashley/test_forum_lti_view.py

```diff
@@ -100,6 +102,91 @@ def test_post_with_an_inactive_user(self):
+            "lti_version": "LTI-1p0",
+            "resource_link_id": "aaa",
+            "context_id": context_id,
+            "lis_person_contact_email_primary": "ashley@example.com",
+            "lis_person_sourcedid": "testuser",
+            "launch_presentation_locale": "en",
+            "roles": "Instructor",
+        }
+        url = f"http://testserver/lti/forum/{forum_uuid}"
+        # we sign the request
+        signed_parameters_context1 = sign_parameters(passport, lti_parameters, url)
+
+        response = self.client.post(
+            f"/lti/forum/{forum_uuid}",
+            data=urlencode(signed_parameters_context1),
+            content_type=CONTENT_TYPE,
+            follow=True,
+        )
+
+        # A LTIContext and a Forum should have been created
+        context = LTIContext.objects.get(lti_id=context_id, lti_consumer_id=consumer1)
+        forum = Forum.objects.get(lti_id=forum_uuid, lti_contexts__id=context.id)
+        lticontext_count = LTIContext.objects.count()
+        forum_count = Forum.objects.count()
+
+        # The response should be a redirection to the forum URL
+        self.assertRedirects(response, f"/forum/forum/{forum.slug}-{forum.id}/")
+
+        # We use the same lti_parameters except the context
+        context2_id = "course-v1:testschool+login+0002"
+        lti_parameters = {
+            "user_id": "643f1625-f240-4a5a-b6eb-89b317807963",
+            "lti_message_type": "basic-lti-launch-request",
+            "lti_version": "LTI-1p0",
+            "resource_link_id": "aaa",
+            "context_id": context2_id,
+            "lis_person_contact_email_primary": "ashley@example.com",
+            "lis_person_sourcedid": "testuser",
+            "launch_presentation_locale": "en",
+            "roles": "Instructor",
+        }
+        # We request the same LTI launch source url
+        signed_parameters_context2 = sign_parameters(passport, lti_parameters, url)
+        response = self.client.post(
+            f"/lti/forum/{forum_uuid}",
+            data=urlencode(signed_parameters_context2),
+            content_type=CONTENT_TYPE,
+            follow=True,
+        )
+        forum_count += 1
+        lticontext_count += 1
+        # A new lti context should have been created
+        self.assertEqual(LTIContext.objects.count(), lticontext_count)
+        # A new forum should have been created
+        self.assertEqual(Forum.objects.count(), forum_count)
+
+        # we resign the request for context1
+        response = self.client.post(
+            f"/lti/forum/{forum_uuid}",
+            data=urlencode(signed_parameters_context1),
+            content_type=CONTENT_TYPE,
+            follow=True,
+        )
+        # No new lti context should have been created
+        self.assertEqual(LTIContext.objects.count(), lticontext_count)
+        # No new forum should have been created
+        self.assertEqual(Forum.objects.count(), forum_count)
+
```

Comment:
I am indeed self.assertRedirects(response, f"/forum/forum/{forum.slug}-{forum.id}/")
I did it for the first one, do you say to repeat the test for each ones ? I can certainly do so, I thought it was redondant and testing something already tested.

Reply:
Oops sorry, I didn't see the first one.

I don't think it's redundant: since there are 2 forums with the same `lti_id`, it's important to check that redirected to the right forum IMO.